Fix for issue #44

oconnor-sn opened the pull request:
Commit based on PR #50 that was never merged.
Added code to incorporate functionality for attachments with signer
requests

oconnor-sn committed:
Fix for issue #44

Commit based on PR #50 that was never merged.
Added code to incorporate functionality for attachments with signer
requests

oconnor-sn committed:
Fix for issue #44

Commit based on PR #50 that was never merged.
Added code to incorporate functionality for attachments with signer
requests

martinytodorov commented on hellosign_sdk/hsclient.py:
```diff
@@ -355,7 +355,7 @@ def send_signature_request(self, test_mode=False, client_id=None, files=None, fi
             signers=None, cc_email_addresses=None, form_fields_per_document=None,
             use_text_tags=False, hide_text_tags=False, custom_fields=None,
-            metadata=None, allow_decline=False, allow_reassign=False, signing_options=None):
+            metadata=None, allow_decline=False, allow_reassign=False, signing_options=None, attachments=False):
```
```suggestion
            metadata=None, allow_decline=False, allow_reassign=False, signing_options=None, attachments=None):
```

martinytodorov commented on hellosign_sdk/hsclient.py:
```diff
@@ -409,6 +409,12 @@ def send_signature_request(self, test_mode=False, client_id=None, files=None, fi
             signing_options (dict, optional): Allows the requester to specify the types allowed for creating a signature. Defaults to account settings.
 
+            attachments (list of dict):            A list of attachments, which each has the following attributes:
+                name (str):                        The name of attachment
```
```suggestion
                name (str):                        The name of the attachment
```

martinytodorov commented on hellosign_sdk/hsclient.py:
```diff
@@ -409,6 +409,12 @@ def send_signature_request(self, test_mode=False, client_id=None, files=None, fi
+            attachments (list of dict):            A list of attachments, which each has the following attributes:
+                name (str):                        The name of attachment
+                instructions (str):                The instructions for uploading the attachment
+                signer_index (int):                The signer's index whose needs to upload the attachments, see signers parameter for more details
```
```suggestion
                signer_index (int):                The index of the signer who needs to upload the attachments, see signers parameter for more details
```

martinytodorov commented on hellosign_sdk/hsclient.py:
```diff
@@ -409,6 +409,12 @@ def send_signature_request(self, test_mode=False, client_id=None, files=None, fi
 
             signing_options (dict, optional): Allows the requester to specify the types allowed for creating a signature. Defaults to account settings.
 
+            attachments (list of dict):            A list of attachments, which each has the following attributes:
```
```suggestion
            attachments (list of dict):            A list of attachments, each with the following attributes:
```

martinytodorov commented on hellosign_sdk/hsclient.py:
```diff
@@ -674,6 +681,13 @@ def send_signature_request_embedded(self, test_mode=False, client_id=None,
 
             signing_options (dict, optional): Allows the requester to specify the types allowed for creating a signature. Defaults to account settings.
 
+            attachments (list of dict):            A list of attachments, which each has the following attributes:
```
Can you update this block with the suggestions from `send_signature_request`?

martinytodorov replied:
@oconnor-sn can you address this comment too please. It should be
`A list of attachments, each with the following attributes:`

oconnor-sn replied:
The version I have with the latest accepted suggestions show that this has been updated.
I reviewed these changes and all occurrences match against your suggestions.

martinytodorov commented on hellosign_sdk/hsclient.py:
```diff
@@ -1906,6 +1921,13 @@ def _send_signature_request(self, test_mode=False, client_id=None, files=None,
 
             is_for_embedded_signing (bool): send_signature_request and send_signature_request_embedded share the same sending logic. To differenciate the two calls embedded requests are now flagged.
 
+            attachments (list of dict):            A list of attachments, which each has the following attributes:
```
Can you update this block with the suggestions from send_signature_request?

martinytodorov replied:
@oconnor-sn can you address this please. It should be `A list of attachments, each with the following attributes:`

martinytodorov reviewed (changes requested):
Thanks for adding the attachments parameter. I left a few comments, mostly typos.

oconnor-sn committed:
Update hellosign_sdk/hsclient.py

Co-Authored-By: Martin Todorov <martin.todorov@hellosign.com>

oconnor-sn committed:
Update hellosign_sdk/hsclient.py

Co-Authored-By: Martin Todorov <martin.todorov@hellosign.com>

oconnor-sn committed:
Update hellosign_sdk/hsclient.py

Co-Authored-By: Martin Todorov <martin.todorov@hellosign.com>

oconnor-sn committed:
Update hellosign_sdk/hsclient.py

Co-Authored-By: Martin Todorov <martin.todorov@hellosign.com>

oconnor-sn commented:
@martinytodorov - please verify updates.

oconnor-sn committed:
Merge remote-tracking branch 'origin/v3' into v3

# Conflicts:
#	hellosign_sdk/hsclient.py

oconnor-sn commented:
Amended the commit and re-pushed @martinytodorov

martinytodorov commented on hsclient.py:
```diff
@@ -0,0 +1,2441 @@
+from hellosign_sdk.utils import HSRequest, HSException, NoAuthMethod, HSAccessTokenAuth, HSFormat, api_resource, api_resource_list
```
@oconnor-sn I think you accidentally added this file. I don't think it should be part of this PR. Can you please remove it.

oconnor-sn replied:
@martinytodorov - Thanks Martin - changes were made to hsclient.py which is why I added this file - please review and advise further.

oconnor-sn committed:
Delete hsclient.py

Add in error

martinytodorov commented on hellosign_sdk/hsclient.py:
```diff
@@ -674,6 +681,13 @@ def send_signature_request_embedded(self, test_mode=False, client_id=None,
 
             signing_options (dict, optional): Allows the requester to specify the types allowed for creating a signature. Defaults to account settings.
 
+            attachments (list of dict):            A list of attachments, which each has the following attributes:
```
```suggestion
            attachments (list of dict):            A list of attachments, each with the following attributes::
```

martinytodorov commented on hellosign_sdk/hsclient.py:
```diff
@@ -1906,6 +1921,13 @@ def _send_signature_request(self, test_mode=False, client_id=None, files=None,
 
             is_for_embedded_signing (bool): send_signature_request and send_signature_request_embedded share the same sending logic. To differenciate the two calls embedded requests are now flagged.
 
+            attachments (list of dict):            A list of attachments, which each has the following attributes:
```
```suggestion
            attachments (list of dict):            A list of attachments, each with the following attributes::
```

oconnor-sn committed:
Update hellosign_sdk/hsclient.py

Co-Authored-By: Martin Todorov <martin.todorov@hellosign.com>

oconnor-sn committed:
Update hellosign_sdk/hsclient.py

Co-Authored-By: Martin Todorov <martin.todorov@hellosign.com>

martinytodorov commented on hellosign_sdk/hsclient.py:
```diff
@@ -1906,6 +1921,13 @@ def _send_signature_request(self, test_mode=False, client_id=None, files=None,
 
             is_for_embedded_signing (bool): send_signature_request and send_signature_request_embedded share the same sending logic. To differenciate the two calls embedded requests are now flagged.
 
+            attachments (list of dict):            A list of attachments, each with the following attributes::
```
```suggestion
            attachments (list of dict):            A list of attachments, each with the following attributes:
```

martinytodorov replied:
Sorry I must have added an additional column.

martinytodorov commented on hellosign_sdk/hsclient.py:
```diff
@@ -674,6 +681,13 @@ def send_signature_request_embedded(self, test_mode=False, client_id=None,
 
             signing_options (dict, optional): Allows the requester to specify the types allowed for creating a signature. Defaults to account settings.
 
+            attachments (list of dict):            A list of attachments, each with the following attributes::
```
```suggestion
            attachments (list of dict):            A list of attachments, each with the following attributes:
```

martinytodorov replied:
Sorry I must have added an additional column.

martinytodorov commented:
Hey @oconnor-sn I checked out the branch on my machine and made a few tests. I realized that there are couple of more things that need to be fixed.

1. You added the `attachments` to the function `_send_signature_request_with_template`. You should remove it from there.
2. Although you added `attachments` to `_send_signature_request`, you are not using them in the payload for the request.

Here's some tips you should follow when opening the PRs:
1. Always test you changes. If you are adding attachments to **signature_request/send** and **signature_request/create_embedded**, then you should use the SDK and make test requests to make sure that your changes are working. Here's [my test request](https://gist.github.com/martinytodorov/d94dc1f6a1b9d2973a0683793bd167eb) if you need one.
2. Use an IDE that can highlight problems for you. That first problem mentioned above is something that my IDE caught. I am using VSCode.
![image](https://user-images.githubusercontent.com/5895193/72375721-7784c780-36c1-11ea-8b2f-cceadf0d8747.png)

Let me know if you have any questions.

oconnor-sn committed:
Updated attachments code

oconnor-sn committed:
Fix for issue #44

Commit based on PR #50 that was never merged.
Added code to incorporate functionality for attachments with signer
requests

oconnor-sn committed:
Fix for issue #44

Commit based on PR #50 that was never merged.
Added code to incorporate functionality for attachments with signer
requests

martinytodorov commented on hellosign_sdk/hsclient.py:
```diff
@@ -2395,6 +2405,7 @@ def _create_embedded_template_draft(self, client_id, signer_roles, test_mode=Fal
         if (merge_fields is not None):
             data.update(merge_fields_payload)
         data = HSFormat.strip_none_values(data)
+        data.update(attachments_payload)
```
Can we move this line with the rest of the updates above.

oconnor-sn commented:
@martinytodorov - anything outstanding on this PR?

martinytodorov reviewed (approved):
Looks good.
